core/src/main/java/de/bwaldvogel/mongo/bson/BsonTimestamp.java

```diff
@@ -2,11 +2,12 @@
 
 import java.time.Instant;
 
-public class BsonTimestamp implements Bson {
+public class BsonTimestamp implements Bson, Comparable<BsonTimestamp> {
 
     private static final long serialVersionUID = 1L;
 
     private long timestamp;
+    private int increment;
```

Comment:
I'm still puzzled why increment should be added on `BsonTimestamp` and not sourced from some "Sequencer". 

Does increment mean nanos part of millis ?

Reply:
OK just read [the documentation](https://docs.mongodb.com/manual/reference/bson-types/#timestamps)